airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -494,9 +494,114 @@ def validate_start_time(self, requested_start_time: int, value: int = 1) -> int:
+    def parent_stream(self):
+        return self._parent_stream
+
+    @parent_stream.setter
+    def parent_stream(self, stream):
+        self._parent_stream = stream
+
+    def _get_parent_stream(self, stream_state: Mapping[str, Any]) -> SourceZendeskSupportStream:
+        return self._stateful_ticket_metrics if stream_state else self._stateless_ticket_metrics
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        return self.parent_stream.get_updated_state(current_stream_state=current_stream_state, latest_record=latest_record)
+
+    def stream_slices(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        self.parent_stream = self._get_parent_stream(stream_state)
+        yield from self.parent_stream.stream_slices(sync_mode=sync_mode, cursor_field=cursor_field, stream_state=stream_state)
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: Optional[List[str]] = None,
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        stream_state: Optional[Mapping[str, Any]] = None,
+    ) -> Iterable[StreamData]:
+        stream_state = self._state_migrator.migrate(stream_state)
+        yield from self.parent_stream.read_records(sync_mode, cursor_field, stream_slice, stream_state)
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        yield self.parent_stream.parse_response(response=response, stream_state=stream_state)
+
+
+class StatelessTicketMetrics(FullRefreshZendeskSupportStream):
+
+    response_list_name: str = "ticket_metrics"
+    most_recently_updated_record: Mapping[str, Any] = None
+    cursor_field: str = "updated_at"
+    state_cursor_field: str = "_ab_updated_at"
+
+    def path(self, **kwargs) -> str:
+        return "ticket_metrics"
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        records = response.json().get(self.response_list_name) or []
+
+        for record in records:
+            updated_at = record[self.cursor_field]
+            record[self.state_cursor_field] = self.str2unixtime(updated_at)
+            if updated_at > self._start_date:
+                if not self.most_recently_updated_record or updated_at > self.most_recently_updated_record[self.cursor_field]:
+                    self.most_recently_updated_record = record
+                yield record
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if self._ignore_pagination:
+            return None
+
+        meta = response.json().get("meta", {}) if response.content else {}
+        return {"page[after]": meta.get("after_cursor")} if meta.get("has_more") else None
+
+    def get_updated_state(
```

Comment:
The way you've designed the legacy `get_updated_state()` method looks more like how we designed the modern `state` setter/getter methods. The modern pattern being that methods like `parse_response()` and `read_records()` invoke `self.state = blah blah` to constantly update state which is then fetched and checkpointed in `core.py`. This looks to be what you're doing with the `self.most_recently_updated_record`.

The more up to date way to manage state would be to:

- Define `state` setter and getter methods
- The `state` getter just returns `self._state`
- The `state` setter` sets the value, but most of the logic below is in a private helper method

It's a subtle difference to use the dedicated getter/setter naming and methods, but it helps us avoid using what is a interface that we _eventually_ intend to deprecate. You can look `source-github` for examples of how we use the setter/getter pattern or happy to sync w/ you on this and see if there's a path forward without using the legacy `get_updated_state()` method if we can avoid it

Reply:
Thanks for tip. I wasn't quite sure what the "new" process was for handling state but I saw notes stating that `get_updated_state` was a legacy method. I've updated to use state getter/setter/private helper.